akka-stream/src/main/scala/akka/stream/scaladsl/Queue.scala

```diff
@@ -32,6 +32,23 @@ trait SourceQueue[T] {
 }
 
 /**
+ * This trait adds completion support to [[SourceQueue]].
+ */
+trait SourceQueueWithComplete[T] extends SourceQueue[T] {
```

Comment:
Yes: the trait is public, so adding methods to it would not in general be binary compatible.